lib/rubocop/cop/style/indent_heredoc.rb

```diff
@@ -87,14 +120,27 @@ def correct_by_library(node)
           end
         end
 
+        def check_style!
+          case style
+          when nil
+            raise Warning, "Auto Correction does not work for #{cop_name}. " \
+                           'Please configure EnforcedStyle.'
+          when :squiggly
+            if target_ruby_version < 2.3
+              raise Warning, '`squiggly` style is selectable only Ruby 2.3 ' \
+                             "or higher for #{cop_name}."
```

Comment:
Maybe we should add some methods for cop specific warnings so we wouldn't have to come up with solutions like this one.